src/java.base/share/classes/jdk/internal/util/ByteArray.java

```diff
@@ -25,400 +25,246 @@
 
 package jdk.internal.util;
 
-import java.lang.invoke.MethodHandles;
-import java.lang.invoke.VarHandle;
-import java.nio.ByteOrder;
+import static jdk.internal.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+import static jdk.internal.util.ArraysSupport.U;
 
 /**
- * Utility methods for packing/unpacking primitive values in/out of byte arrays
- * using {@linkplain ByteOrder#BIG_ENDIAN big endian order} (aka. "network order").
+ * Utility methods for packing/unpacking primitive values in/out of byte arrays.
  * <p>
- * All methods in this class will throw an {@linkplain NullPointerException} if {@code null} is
- * passed in as a method parameter for a byte array.
+ * All methods perform checked access, throwing NPE for null arrays or AIOOBE
+ * if the start index of the access will cause out of bounds read/write.
+ * <p>
+ * Methods are grouped into 4 categories: BO for explicit byte order, BE for
+ * big endian, LE for little endian, and there are boolean access methods that
+ * does not depend on byte order.
+ * <p>
+ * Types supported including non-byte primitives numbers (char, short, int,
+ * float, long, double), raw float/double writes with special NaN treatment,
+ * unsigned shorts (u2), and single-byte boolean checking != 0.
  */
 public final class ByteArray {
 
     private ByteArray() {
     }
 
-    private static final VarHandle SHORT = create(short[].class);
-    private static final VarHandle CHAR = create(char[].class);
-    private static final VarHandle INT = create(int[].class);
-    private static final VarHandle FLOAT = create(float[].class);
-    private static final VarHandle LONG = create(long[].class);
-    private static final VarHandle DOUBLE = create(double[].class);
-
-    /*
-     * Methods for unpacking primitive values from byte arrays starting at
-     * a given offset.
-     */
-
-    /**
-     * {@return a {@code boolean} from the provided {@code array} at the given {@code offset}}.
-     *
-     * @param array  to read a value from.
-     * @param offset where extraction in the array should begin
-     * @throws IndexOutOfBoundsException if the provided {@code offset} is outside
-     *                                   the range [0, array.length - 1]
-     * @see #setBoolean(byte[], int, boolean)
-     */
+    // Basic types with Byte Order
+
+    public static char getCharBO(byte[] array, int index, boolean big) {
+        Preconditions.checkIndex(index, array.length - Character.BYTES + 1, Preconditions.AIOOBE_FORMATTER);
+        return U.getCharUnaligned(array, ARRAY_BYTE_BASE_OFFSET + index, big);
+    }
+
+    public static short getShortBO(byte[] array, int index, boolean big) {
```

Comment:
If we have methods `getShortBE` and `getShortLE` then perhaps this method should just be called `getShort`.